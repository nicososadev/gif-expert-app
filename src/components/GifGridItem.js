import React from 'react'
import propTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {
    
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propType = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}
