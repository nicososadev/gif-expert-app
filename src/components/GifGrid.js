import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

    const { data:images , loading } = useFetchGifs( category )
    
    return (
        <Fragment>
            <h3>{ category }</h3>

            { loading && <p className="card animate__animated animate__flash">Loading...</p> }

            <div className="card-grid">
            
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
                
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {

    category: PropTypes.string.isRequired

}
