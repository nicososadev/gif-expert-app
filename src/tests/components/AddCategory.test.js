
import React from 'react'
import { AddCategory } from "../../components/AddCategory"
import { shallow } from "enzyme"


describe('Pruebas en el componente AddCategory', () => {

    const setCategories = jest.fn()
    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> )

    beforeEach( () => {
        jest.clearAllMocks()
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> )
    })
    
    test('El componente debe mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot()
    })
    
    test('Debe cambiar el input de texto', () => {
        
        const input = wrapper.find('input')
        const value = 'Texto'

        input.simulate('change', { target: { value } })

        const inputValue = wrapper.find('input').props().value

        expect( inputValue ).toBe( value )

    })

    test('No debe cargar informacion con el evento submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault: () => {} })

        expect( setCategories ).not.toHaveBeenCalled()
    })
    
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        const input = wrapper.find('input')
        const value = 'Texto'

        input.simulate('change', { target: { value } })

        wrapper.find('form').simulate('submit', { preventDefault: () => {} })

        const inputValue = wrapper.find('input').props().value

        expect( setCategories ).toHaveBeenCalled()

        expect( inputValue ).toBe( '' )

    })
    
    
})
