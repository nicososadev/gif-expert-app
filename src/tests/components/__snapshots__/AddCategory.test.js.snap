// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory El componente debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
